ui/e2e/src/tests/panelEditor.spec.ts

```diff
@@ -0,0 +1,43 @@
+// Copyright 2023 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { test, expect } from '../fixtures/dashboardTest';
+import { PanelEditor } from '../pages/PanelEditor';
+
+test.use({
+  dashboardName: 'Panels',
+});
+
+test.describe('Dashboard: Panel Editor', () => {
+  test('should show discard changes confirmation dialog', async ({ dashboardPage }) => {
+    await dashboardPage.startEditing();
+    await dashboardPage.addPanel();
+    await dashboardPage.panelEditor.isVisible();
+    const panelEditor = new PanelEditor(dashboardPage.panelEditor);
```

Comment:
Could you use the `isVisible` on the panel editor class here. It's a bit smarter about waiting for all the animations to finish on the panel sliding in.

Related, it might be nice to create a `getPanelEditor` helper in `DashboardPage` that simplifies the `new PanelEditor` bit for you. I added a similar helper like that for `getVariableEditor`, but didn't have a chance to go back and refactor a similar change for the panel editor. If you don't feel like doing this, no worries, and I'll come back to it at some point.